Retire Queue submissions when Timeline Semaphore is waited on

farnoy opened the pull request:
This addresses #1502 and fixes validation warnings in the CTS tests I modified (linked in that issue).

cc @jasuarez, you were the main contributor to timeline semaphores, I would appreciate a review

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KhronosGroup/Vulkan-ValidationLayers?pullRequest=1511) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/Vulkan-ValidationLayers?pullRequest=1511) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KhronosGroup/Vulkan-ValidationLayers?pullRequest=1511) it.</sub>

jasuarez commented:
@farnoy i'm reviewing it; please, do not forget to sign the CLA

jasuarez commented:
Can you squash all the changes in a single commit? Most of the commits are just fixes to the first commit

farnoy commented:
@jasuarez Done and done.

jasuarez commented on layers/state_tracker.cpp:
```diff
@@ -1595,18 +1596,23 @@ void ValidationStateTracker::PostCallRecordQueueSubmit(VkQueue queue, uint32_t s
                     } else {
-                        pSemaphore->payload = timeline_semaphore_submit->pSignalSemaphoreValues[i];
+                        uint64_t payload = timeline_semaphore_submit->pSignalSemaphoreValues[i];
+                        pSemaphore->payload = payload;
```
This is not correct; we can't update the semaphore payload because the semaphore has not been signaled yet, and we need the current payload for different validation cases. See https://github.com/jasuarez/Vulkan-ValidationLayers/commit/94c3c663a93e76ee19e62de1f7dba0cc3320174f for a fix.

The update payload should be done after the semaphore is signaled. I have this update done in a follow up commit (https://github.com/jasuarez/Vulkan-ValidationLayers/commit/d318f24fb395990e4326f4aaf180e7f7f1eb9719)

jasuarez commented on layers/state_tracker.cpp:
```diff
@@ -1869,6 +1879,29 @@ void ValidationStateTracker::PostCallRecordWaitForFences(VkDevice device, uint32
+            QUEUE_STATE &queueState = pair.second;
+            for (const auto &submission : queueState.submissions) {
+                for (const auto &signalSemaphore : submission.signalSemaphores) {
+                    if (signalSemaphore.payload <= until_payload) RetireWorkOnQueue(&queueState, signalSemaphore.seq);
```
Missing to check the signalSemaphore is the semaphore we want to retire (see https://github.com/jasuarez/Vulkan-ValidationLayers/commit/94c3c663a93e76ee19e62de1f7dba0cc3320174f for a fix)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/KhronosGroup/Vulkan-ValidationLayers?pullRequest=1511) <br/>All committers have signed the CLA.

farnoy commented:
@jasuarez Thanks for the feedback! I cherry-picked your fix commit https://github.com/jasuarez/Vulkan-ValidationLayers/commit/94c3c663a93e76ee19e62de1f7dba0cc3320174f , rebased my branch and squashed it into a single commit again. I think this is what you meant. 

If not, the previous commit that you reviewed (without these fixes) is https://github.com/farnoy/Vulkan-ValidationLayers/commit/90a97399614cc9a438ad11a5dd2fc6f42fc21813 .

jasuarez commented:
@mark-lunarg could you take a look?

mark-lunarg commented:
Running through CI.

mark-lunarg commented:
Rebased, tweaked commit msg and pushed to Master.
